core/src/main/scala/org/http4s/syntax/EffectRequestSyntax.scala

```diff
@@ -5,6 +5,7 @@ import cats._
 import cats.implicits._
 
 trait EffectRequestSyntax extends Any {
+  @deprecated("Use map or flatMap directly on the request", "0.18.0-M2")
```

Comment:
This message could also refer to the new headers arguments.

We could be more precise if we'd move the deprecation to the ops class, but then we'd need to override every method, because it inherits a bunch of (undeprecated) concrete implementations.

Reply:
By deprecating here instead of deprecating the syntax type itself, users of `Http4sDsl` or `AllSyntax` can emit deprecation warnings on usage without itself extending a deprecated type (a fatal warning).